lib/platform/git/index.ts

```diff
@@ -81,6 +81,24 @@ async function isDirectory(dir: string): Promise<boolean> {
   }
 }
 
+async function getDefaultBranch(git: Git.SimpleGit): Promise<string> {
```

Comment:
We shoud add a `setGitConfig` where we init the `SimpleGit` instance stored as module scope variable.
That should be called by platform init for now.

After move to `util/gitfs` we can call it from `util/index`:

https://github.com/renovatebot/renovate/blob/e8c80fbe625acf1304ac9da8ca979e82c24e14ec/lib/util/index.ts#L7-L12

Reply:
So, basically module-scoped variables are okay in `util/gitfs`, but not in calling code which have git `Storage` instances for now?

Reply:
Yes, i thougt this pr will eliminate the `Storage` class completly. Do you want to split it in smaller parts?

Reply:
I think single file is good, just asked to clarify about stateful parts

Reply:
Maybe we can leave a small `Storage` wrapper class for first step, so we have less changes in this pr (no need for changing other platform code).

@rarkins What do you think?

Reply:
Should the refactored module be used for only single repo, right?

Reply:
For now we could keep the class with this file but remove its instance from all the other platforms. Most important is that this file should contain all its own state and we no longer have the concept of `config.storage` in each of the other platforms